### ISSUE TITLE
Metrics Processor and Checkpointer APIs are confusingly named

### ISSUE THREAD

**Issue by jmacd** [open-telemetry/opentelemetry-go#1056]
See this comment thread: https://github.com/open-telemetry/opentelemetry-go/pull/1055#discussion_r470215544

One way to resolve this is to remove the name "Checkpointer". We can rename the current Processor to "AccumulatorProcessor" (which is just one method) and rename the current Checkpointer to "Processor" (which is 4 methods). 

This should be addressed after #1055 merges.

**Comment by MrAlias** [comment]
@jmacd is this "needed for GA"?

**Comment by jmacd** [comment]
No. I would say this can be easily resolved by renaming things, shouldn't affect GA.

**Comment by MrAlias** [comment]
The [metric SDK redesign project](https://github.com/orgs/open-telemetry/projects/22) will implement a workflow where these types will no longer exist. Closing this as we should not waste effort renaming these when they are going to be discarded.
